<launch>
    <!-- This is the launch file for PhytO-ARM's core processes.
         See arm launch files for payload specific settings. -->

    <!-- Most settings are in config.yaml, which is easier to edit. -->
    <rosparam command="load" file="$(arg config_file)" />

    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch">
        <arg name="use_compression" value="true" />
    </include>

    <node name="lock_manager" pkg="phyto_arm" type="lock_manager.py" />

    <node if="$(arg network_data_capture)" name="network_data_capture" pkg="phyto_arm" type="network_data_capture.py" required="false" />

    <group if="$(arg classifier)">
        <node name="classifier" pkg="triton_classifier" type="classifier_node.py" />

        <node name="image_converter" pkg="image_transport" type="republish"
              args="compressed in:=/ifcb/roi/image raw out:=/ifcb/roi/image" />
    </group>

    <node name="gps" pkg="gpsd_client" type="gpsd_client">
        <remap from="/fix" to="~fix" />
        <remap from="/extended_fix" to="~extended_fix" />
    </node>

    <node name="web" pkg="phyto_arm" type="web_node.py" />

    <group ns="camera">
        <node name="fore_camera" pkg="rtsp_camera" type="rtsp_camera_node">
            <remap from="~image" to="~image/do_not_use" />
            <remap from="~image/do_not_use/compressed" to="~image/compressed" />
        </node>

        <node name="aft_camera" pkg="rtsp_camera" type="rtsp_camera_node">
            <remap from="~image" to="~image/do_not_use" />
            <remap from="~image/do_not_use/compressed" to="~image/compressed" />
        </node>
    </group>

    <node if="$(arg digital_logger)" name="digital_logger" pkg="dli_power_switch" type="dli_power_switch_node.py"/>
</launch>
